src/main/java/org/jitsi/jicofo/recording/jibri/JibriRecorder.java

```diff
@@ -0,0 +1,484 @@
+     * {@inheritDoc}
+     */
+    @Override
+    synchronized public void processPacket(Packet packet)
+    {
+        JibriIq iq = (JibriIq) packet;
+
+        String from = iq.getFrom();
+
+        if (logger.isDebugEnabled())
+            logger.debug("Got Jibri packet: " + packet.toXML());
+
+        if (recorderComponentJid != null &&
+            (from.equals(recorderComponentJid) ||
+
+            (from +"/").startsWith(recorderComponentJid)))
+        {
+            processJibriIqFromJibri(iq);
+        }
+        else
+        {
+            String roomName = MucUtil.extractRoomNameFromMucJid(from);
+            if (roomName == null)
+            {
+                return;
+            }
+
+            if (!conference.getRoomName().equals(roomName))
+            {
+                logger.debug(
+                        "Ignored packet from: " + roomName
+                            + ", my room: " + conference.getRoomName()
+                            + " p: " + packet.toXML());
+                return;
+            }
+
+            XmppChatMember chatMember
+                = conference.getChatRoom().findChatMember(from);
+            if (chatMember == null)
+            {
+                logger.error("ERROR chat member not found for: " + from);
+                return;
+            }
+
+            processJibriIqFromMeet(iq, chatMember);
+        }
+    }
+
+    private void processJibriIqFromMeet(JibriIq iq, XmppChatMember sender)
+    {
+        JibriIq.Action action = iq.getAction();
+
+        if (JibriIq.Action.UNDEFINED.equals(action))
+            return;
+
+        String senderMucJid = sender.getContactAddress();
+
+        logger.debug(
+                "Jibri request from " + senderMucJid + " iq: " + iq.toXML());
+
+        // start ?
+        if (JibriIq.Action.START.equals(action) &&
+            JibriIq.Status.OFF.equals(jibriStatus) &&
+            recorderComponentJid == null)
+        {
+            if (!verifyModeratorRole(iq))
+            {
+                logger.warn(
+                        "Ignored Jibri request from non-moderator: "
+                            + senderMucJid);
+                return;
+            }
+
+            // Check if we have Jibri available
+            String jibriJid = jibriDetector.selectJibri();
+            if (jibriJid == null)
+            {
+                sendErrorResponse(
+                    iq, XMPPError.Condition.service_unavailable, null);
+                return;
+            }
+
+            JibriIq startIq = new JibriIq();
+            startIq.setTo(jibriJid);
+            startIq.setType(IQ.Type.SET);
+            startIq.setAction(JibriIq.Action.START);
+
+            startIq.setStreamId(iq.getStreamId());
+            startIq.setUrl(iq.getUrl());
+
+            logger.debug("Starting Jibri recording: " + startIq.toXML());
+
+            IQ startReply
+                = (IQ) xmpp.getXmppConnection()
+                        .sendPacketAndGetReply(startIq);
```

Comment:
If I am reading this correctly, the Jibri instance for jibriJid will remain with status "IDLE" until we receive a response from the network. And if we receive another request for recording during this time, the same Jibri instance will be selected again since it's on top of the list.